docs/operator-manual/config-management-plugins.md

```diff
@@ -201,18 +207,172 @@ volumes:
     2. Make sure that sidecar container is running as user 999.
     3. Make sure that plugin configuration file is present at `/home/argocd/cmp-server/config/plugin.yaml`. It can either be volume mapped via configmap or baked into image.
 
+#### Register the plugin as a service
+
+Argo CD needs to have access to the cluster to discover services. To do this it uses a standard Kubernetes Service Account token to authenticate
+with the cluster. It also needs a Role or ClusterRole bound to that Service Account to access with the permissions get, list and watch the namespace(s)
+that the services are in. By default (no configuration) the services are looked for in the same namespace as the argocd-repo-server. You can set
+`ARGOCD_SERVICE_PLUGINS_NAMESPACE` to the name of a namespace to change it to look in a different namespace. To look in all
+namespaces set `ARGOCD_SERVICE_PLUGINS_NAMESPACE` to `*`. You cannot configure multiple specific namespaces, just one namespace or all namespaces.
+
+!!! Ensure that:
+    1. The argocd-repo-server deployment has `automountServiceAccountToken: true`
+	2. The argocd-repo-server's service account has a role binding allowing `get`, `list` and `watch` on services.
+
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: argocd-repo-server-getsvcs
+  namespace: argocd
+rules:
+- apiGroups:
+    - ""
+  resources:
+    - services
+  verbs:
+    - get
+    - list
+    - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: argocd-repo-server-getsvcs
+  namespace: argocd
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: argocd-repo-server-getsvcs
+subjects:
+- kind: ServiceAccount
+  name: argocd-repo-server
+  namespace: argocd
+```
+
+or for cluster wide
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: argocd-repo-server-getsvcs
+rules:
+- apiGroups:
+    - ""
+  resources:
+    - services
+  verbs:
+    - get
+    - list
+    - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: argocd-repo-server-getsvcs
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: argocd-repo-server-getsvcs
+subjects:
+- kind: ServiceAccount
+  name: argocd-repo-server
+  namespace: argocd
+```
+
+To install a plugin as a service, deploy something running your code. This would usually be a deployment, but Argo CD doesn't
+actually care what is behind the service. You can use either an off-the-shelf or custom-built plugin image as sidecar image. This
+should follow the same rules as for a sidecar.
+
+To advertise the service to Argo CD you must provide a service with the label `argocd.argoproj.io/plugin: true`. Each port in the
+service will be treated as a separate plugin, and the name of the port is the name of the plugin, which will be used by applications
+which specify their plugin by name.
```

Comment:
```suggestion
To advertise the service to Argo CD you must provide a service with the label `argocd.argoproj.io/plugin: true`.
Each port in the service will be treated as a separate plugin.
The name of the port is used as the name of the plugin.
Applications can specify a plugin by the plugin name.
```

Reply:
Good suggestion - I've modified it slightly but take it in.